src/tools/pub/dependencies.md

```diff
@@ -442,6 +439,32 @@ To identify dependencies in your app or package that
 aren't on the latest stable versions,
 use [`dart pub outdated`][].
 
+### Use tight dev_dependencies
+
+Dependencies can be loose, specifying dependency at the highest version digit
+like `^2.0.0`, or tight, specifying dependency at the lowest version digit like
+`^1.2.4`.
+
+In your `pubspec.yaml` file, follow these recommendations.
+
+* Set regular `dependencies` loose to increase the chance of a version solver success.
```

Comment:
This section's title only mentions dev dependencies. Rather than expanding the focus of this section, I'd consider dropping this as the best practices for normal dependencies are much more nuanced and very situation dependent.

That's not to say we shouldn't have guidelines for them, but if we do have guidelines, we have to make them very clear.